File: product-processor-service/src/main/java/com/example/product_processor_service/configuration/kafka/template/KafkaTemplateConfiguration.java
package com.example.product_processor_service.configuration.kafka.template;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.kafka.core.ProducerFactory;

@Configuration
public class KafkaTemplateConfiguration {
    @Bean
    public KafkaTemplate<String, String> stringKafkaTemplate(ProducerFactory<String, String> producerFactory) {
        return new KafkaTemplate<>(producerFactory);
    }
}

File: product-manager-service/src/main/kotlin/com/example/product_manager_service/configuration/kafka/template/KafkaTemplateConfiguration.kt
package com.example.product_manager_service.configuration.kafka.template

import com.example.product_manager_service.model.product.dto.ProductPublisherDto
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.kafka.core.ProducerFactory

@Configuration
class KafkaTemplateConfiguration() {
    @Bean
    fun productPublisherKafkaTemplate(producerFactory: ProducerFactory<String, ProductPublisherDto>): KafkaTemplate<String, ProductPublisherDto> =
        KafkaTemplate(producerFactory)
}